Startup.cs

```diff
@@ -0,0 +1,112 @@
+using Lombiq.Privacy.Activities;
+using Lombiq.Privacy.Constants;
+using Lombiq.Privacy.Drivers;
+using Lombiq.Privacy.Filters;
+using Lombiq.Privacy.Handlers;
+using Lombiq.Privacy.Migrations;
+using Lombiq.Privacy.Models;
+using Lombiq.Privacy.Navigation;
+using Lombiq.Privacy.Permissions;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Display.ContentDisplay;
+using OrchardCore.Data.Migration;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.Modules;
+using OrchardCore.Navigation;
+using OrchardCore.ResourceManagement;
+using OrchardCore.Security.Permissions;
+using OrchardCore.Settings;
+using OrchardCore.Users.Events;
+using OrchardCore.Users.Models;
+using OrchardCore.Users.Services;
+using OrchardCore.Workflows.Helpers;
+using System;
+
+namespace Lombiq.Privacy
+{
+    [Feature(FeatureNames.Module)]
+    public class Startup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.AddScoped<IPermissionProvider, PrivacyConsentPermissions>();
+            services.AddScoped<INavigationProvider, AdminMenu>();
+            services.AddTransient<IConfigureOptions<ResourceManagementOptions>, ResourceManagementOptionsConfiguration>();
+        }
+    }
+
+    [Feature(FeatureNames.ConsentBanner)]
+    public class ConsentBannerStartup : StartupBase
+    {
+        public override void Configure(
+            IApplicationBuilder app,
+            IEndpointRouteBuilder routes,
+            IServiceProvider serviceProvider) =>
+                app.UseCookiePolicy();
+
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<CookiePolicyOptions>(options =>
+            {
+                options.CheckConsentNeeded = context => true;
+                options.MinimumSameSitePolicy = SameSiteMode.None;
+                options.ConsentCookie.Expiration = new TimeSpan(30, 0, 0, 0);
+            });
+
+            services.Configure<MvcOptions>((options) =>
+                options.Filters.Add(typeof(ConsentBannerInjectionFilter)));
+            services.AddScoped<IDisplayDriver<ISite>, ConsentBannerSettingsDisplayDriver>();
+        }
+    }
+
+    [Feature(FeatureNames.RegistrationConsent)]
+    public class RegistrationConsentStartup : StartupBase
+    {
+        public override void ConfigureServices(IServiceCollection services)
+        {
+            services.Configure<CookiePolicyOptions>(options =>
+            {
+                options.CheckConsentNeeded = context => IsConsentNeeded(context);
+                options.MinimumSameSitePolicy = SameSiteMode.None;
+                options.ConsentCookie.Expiration = new TimeSpan(30, 0, 0, 0);
```

Comment:
Biztos, hogy ez mindenképp később fut le, mint a `ConsentBannerStartup` és így felülírja, ha mindkettő be van kapcsolva?

Reply:
Teszteltem különböző kombinációkban és jól müködött(a különböző esetek között töröltem a db-t is) de a biztonság kedvéért ezeket átrakom a `Lombiq.Privacy` `Startup`-jába és akkor csak egyhelyen lesz ez a config